File: Math/codeforces_1841B.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/1841/B
// submission link ===>> https://codeforces.com/contest/1841/submission/209974072

#include <iostream>
using namespace std;

int main()
{
    int t;
    cin >> t;

    while (t--)
    {
        int n;
        cin >> n;
        int flag = 0;
        int first = -1;
        int last = -1;

        for (int i = 0; i < n; i++)
        {
            int num;
            cin >> num;

            if (flag)
            {
                if (num >= last && num <= first)
                {
                    last = num;
                    cout << 1;
                }
                else
                {
                    cout << 0;
                }
            }
            else
            {
                if (first == -1)
                {
                    first = num;
                    last = num;
                    cout << 1;
                }
                else if (num >= last)
                {
                    last = num;
                    cout << 1;
                }
                else if (first >= num)
                {
                    last = num;
                    flag = 1;
                    cout << 1;
                }
                else
                {
                    cout << 0;
                }
            }
        }

        cout << endl;
    }

    return 0;
}

File: Math/codeforces_1841B.py

# Problem link ===>> https://codeforces.com/problemset/problem/1841/B

t = int(input())

for _ in range(t):
    n = int(input())
    flag = 0
    first = -1
    last = -1

    for _ in range(n):
        num = int(input())

        if flag:
            if num >= last and num <= first:
                last = num
                print(1, end="")
            else:
                print(0, end="")
        else:
            if first == -1:
                first = num
                last = num
                print(1, end="")
            elif num >= last:
                last = num
                print(1, end="")
            elif first >= num:
                last = num
                flag = 1
                print(1, end="")
            else:
                print(0, end="")

    print()
